homeassistant/components/shelly/__init__.py

```diff
@@ -260,6 +261,12 @@ async def _async_setup_rpc_entry(hass: HomeAssistant, entry: ShellyConfigEntry)
         runtime_data.platforms = PLATFORMS
         try:
             await device.initialize()
+
+            # Update entry data with script support info
+            data = {**entry.data}
+            data[CONF_SCRIPT] = await device.supports_scripts()
```

Comment:
Should we only check scripting support if there is no `CONF_SCRIPT` in the config entry data?

Reply:
We never add CONF_SCRIPT to the entry as we prefer to get it dynamically at each `init` in case of future firmware changes.